src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnRpService.java

```diff
@@ -210,23 +222,32 @@ private String generateNewKey() {
                     }
                 }
                 if (toUpdate.isEmpty()) {
-                    return Optional.empty();
+                    throw new WebAuthnAuthenticationException(account.getSubject(),
+                            "Could not find the requested credential in the account");
                 }
                 WebAuthnCredential credential = toUpdate.get();
                 credentials.remove(credential);
                 credential.setSignatureCount(result.getSignatureCount());
                 credential.setLastUsedOn(new Date());
                 webAuthnCredentialsRepository.save(credential);
-                return Optional.of(account.getUsername());
+                final String username = account.getUsername();
+                if (username != null) {
```

Comment:
rewrite this check. Favour the form with a single block
if(error condition) { throw exception}

Reply:
Fixed in ce20a37ae74e0fa905068dac31715f9fbb01d9f9